### PR TITLE
Metro mode selector: coreutils

### PR DESCRIPTION
**Pull request by daniel-heppner-ibigroup**
The core utils changes are mostly focused on generating OTP GraphQL Queries to go along with the new mode selector and associated changes. We also have code to generate combinations, so they no longer need to be specified by the consumer.

### CODE REVIEW

**Comment by daniel-heppner-ibigroup** [comment]
Blocked until #551 is merged.

File: packages/core-utils/src/planQuery.graphql
Comment (binh-dam-ibigroup) [inline packages/core-utils/src/planQuery.graphql:21]: Add a comment regarding what setting the locale does.

File: packages/core-utils/src/query-gen.ts
Comment (binh-dam-ibigroup) [inline packages/core-utils/src/query-gen.ts:51]:
```suggestion
        return [...prev, transportMode];
```

File: packages/core-utils/src/query-gen.ts
Comment (binh-dam-ibigroup) [inline packages/core-utils/src/query-gen.ts:76]: Add a comment that this variable maps each mode into a broader travel mode.

File: packages/core-utils/src/query-gen.ts
Comment (binh-dam-ibigroup) [inline packages/core-utils/src/query-gen.ts:138]: Extract a variable, so it's easier to read what this condition is about.

File: packages/core-utils/src/query-gen.ts
Comment (binh-dam-ibigroup) [inline packages/core-utils/src/query-gen.ts:156]:
```suggestion
 * Generates a list of queries for OTP to get a comprehensive
 * ```

File: packages/core-utils/src/query-gen.ts
Comment (binh-dam-ibigroup) [inline packages/core-utils/src/query-gen.ts:178]: You might be able to move this line to the function declaration.

File: packages/core-utils/src/query-gen.ts
Comment (binh-dam-ibigroup) [inline packages/core-utils/src/query-gen.ts:199]: Declare `modes` within line 178.

File: packages/core-utils/package.json
Comment (binh-dam-ibigroup) [inline packages/core-utils/package.json:14]: You will have to update the version of `@opentripplanner/types` under `devDependencies`.

**Review by binh-dam-ibigroup — APPROVED** [review]
Some minor cleanup and a conflict to fix, but otherwise looks good.

File: packages/core-utils/src/__tests__/query-params.ts
Comment (daniel-heppner-ibigroup) [reply packages/core-utils/src/__tests__/query-params.ts:78]: It was test option, but I made it more clear.

**Review by AdrianaCeric — APPROVED** [review]
I have limited knowledge of the new mode selector and OTP flex compatible modes so my review isn't too in-depth here, but code makes sense!

File: packages/core-utils/src/__tests__/query-params.ts
Comment (daniel-heppner-ibigroup) [reply packages/core-utils/src/__tests__/query-params.ts:79]: Thanks for the suggestion, I've added it.

File: packages/core-utils/src/__tests__/query-params.ts
Comment (daniel-heppner-ibigroup) [reply packages/core-utils/src/__tests__/query-params.ts:190]: Miles wrote that, but I think the question is whether using scooter_rent and walk can be used together. (the answer is yes I think)

**Pull request merged**
